File: Core CS/Algorithms/Warmup/Staircase/staircase.kt
import java.io.*
import java.util.*

fun staircase(n: Int): Unit {
    for(i in 1..n) {
        //repeat strings using CharSequence.repeat(n: Int): String
        print(" ".repeat(n-i))
        println("#".repeat(i))
    }
}

fun main(args: Array<String>) {
    val scan = Scanner(System.`in`)

    val n = scan.nextLine().trim().toInt()

    staircase(n)
}

File: Core CS/Algorithms/Warmup/Staircase/staircase.scala
import java.io._
import java.util._

object Solution {

    // Complete the staircase function below.
    def staircase(n: Int) {
        for(i <- 1 to n){
            //repeat strings in scala by multiplication
            print(" " * (n-i))
            println("#" * (i))
        }
    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val n = stdin.readLine.trim.toInt

        staircase(n)
    }
}
